### DISCUSSION TITLE
enh: Bind memory to specific models only

### DISCUSSION THREAD

**Discussion by Tycho-S** [open-webui/open-webui#3751] [Ideas]
**Is your feature request related to a problem? Please describe.**
When I create memories, it is currently sent to any and all model I interact with. I don't want to share this stuff with cloud-based models (ChatGPT)

**Describe the solution you'd like**
It would be great if memories could be assigned to specific models only (including custom configurations under Workspaces)

**Describe alternatives you've considered**
Creating a separate account, but it's a bit clumsy

**Additional context**

**Comment by tjbck** [comment]
This should be implemented as Functions! PR welcome!

**Comment by tompre** [comment]
This would be a great improvement! Maybe I could give it a go.

@tjbck What do you mean by implemented as Functions?

Should the Functions maintain their own memories or could the memory interface be extended such that, the memories are gated via an additional key (i.e. the model id[s])?
